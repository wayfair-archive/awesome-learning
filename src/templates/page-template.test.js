import React from 'react';
import renderer from 'react-test-renderer';
import PageTemplate from './page-template';

describe('PageTemplate', () => {
  const props = {
    data: {
      markdownRemark: {
        html: '<p>test</p>',
        frontmatter: {
          title: 'test',
          description: 'test'
        }
      },
      site: {
        siteMetadata: {
          title: 'test',
          subtitle: 'test'
        }
      }
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<PageTemplate {...props} />, {
      createNodeMock: () => ({
        scrollIntoView() {}
      })
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
