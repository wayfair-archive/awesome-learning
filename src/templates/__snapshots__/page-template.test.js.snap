// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate renders correctly 1`] = `
Array [
  <StaticQuery
    render={[Function]}
  />,
  <div
    className="Layout"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "marginLeft": "auto",
          "marginRight": "auto",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "normal",
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 0,
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
            "marginLeft": "-15px",
            "marginRight": "-15px",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 0,
              "left": "auto",
              "marginLeft": "0%",
              "maxWidth": "100%",
              "minHeight": "1px",
              "paddingLeft": "15px",
              "paddingRight": "15px",
              "position": "relative",
              "right": "auto",
              "width": Object {
                "lg": null,
                "md": null,
                "sm": null,
                "xl": null,
                "xs": null,
              },
            }
          }
        >
          <div
            className="Page"
          >
            <div
              className="Page-inner"
            >
              <h1
                className="Page-title"
              >
                test
              </h1>
              <div
                className="Page-body"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<p>test</p>",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        style={
          Object {
            "clear": "both",
            "display": "table",
          }
        }
      />
    </div>
  </div>,
]
`;
